src/AppWindow.cpp

```diff
@@ -28,55 +29,75 @@ AppWindow::AppWindow(BRect frame)
 	MainView = new AppView(Bounds());
 	AddChild(MainView);
 	
-	PassOut = new RandoTextView(BRect(5,25,200,40), "PassOut",
-		BRect(0,0,200,30), be_bold_font, NULL, B_FOLLOW_LEFT|B_FOLLOW_TOP,
+	// Result Label:
+	BStringView *result = new BStringView(BRect(100,30,11,11),"resultlabel","RESULT");
+	result->ResizeToPreferred();
+	MainView->AddChild(result);
+	
+	// Result:
+	PassOut = new RandoTextView(BRect(5,50,245,80), "PassOut",
+		BRect(5,0,245,60), be_bold_font, NULL, B_FOLLOW_LEFT|B_FOLLOW_TOP,
 		B_WILL_DRAW|B_NAVIGABLE);
 	PassOut->MakeEditable(false);
 	MainView->AddChild(PassOut);
-			
-	ParamsBox = new RandoBox(BRect(5,50,200,146), "ParamsBox", "", B_FOLLOW_LEFT|B_FOLLOW_TOP,
+	
+	// Parameters Label:
+	BStringView *param = new BStringView(BRect(90,90,11,11),"paramlabel","Parameters:");
+	param->ResizeToPreferred();
+	MainView->AddChild(param);
+	
+	// Parameters Box:
+	ParamsBox = new RandoBox(BRect(5,100,245,210), "ParamsBox", "", B_FOLLOW_LEFT|B_FOLLOW_TOP,
 		B_WILL_DRAW|B_NAVIGABLE, B_FANCY_BORDER);
 	MainView->AddChild(ParamsBox);
 	
-	PassLength = new BTextControl(BRect(5,5,189,40), "PassLength", "Password length", NULL,
+	// Length Label:
+	BStringView *length = new BStringView(BRect(60,15,100,40),"lengthlabel","Length:");
+	ParamsBox->AddChild(length);
+	
+	// Length TextBox:
+	PassLength = new BTextControl(BRect(70,15,140,40), "PassLength", "Password length", NULL,
 		new BMessage(SEQ_LEN_MSG), B_FOLLOW_LEFT|B_FOLLOW_TOP, B_NAVIGABLE);
```

Comment:
The bug is that the B_WILL_DRAW flag is not included. If you bitwise-or it with the last argument, the label will appear. :)

Reply:
Thanks ! Done :) commit https://github.com/HaikuArchives/Randomizer/pull/1/commits/fd485d3eef51150cc25c956a5247109031c410ba